tests/fixture/testnet/local/network.go

```diff
@@ -0,0 +1,644 @@
+		}
+	}
+	return uris
+}
+
+type NodeStopError struct {
+	NodeID    ids.NodeID
+	StopError error
+}
+
+func (e *NodeStopError) Error() string {
+	return fmt.Sprintf("failed to stop node %s: %v", e.NodeID, e.StopError)
+}
+
+type NetworkStopError struct {
+	Errors []*NodeStopError
+}
+
+func (e *NetworkStopError) Error() string {
+	return fmt.Sprintf("failed to stop network: %v", e.Errors)
+}
+
+// Stop all nodes in the network.
+func (ln *LocalNetwork) Stop() error {
+	errs := []*NodeStopError{}
+	// Assume the nodes are loaded and the pids are current
+	for _, node := range ln.Nodes {
+		if err := node.Stop(); err != nil {
+			errs = append(errs, &NodeStopError{
+				NodeID:    node.NodeID,
+				StopError: err,
+			})
+		}
+	}
+	if len(errs) > 0 {
+		// TODO(marun) When avalanchego updates to go 1.20, update to
+		// use the new capability to wrap multiple errors.
+		return &NetworkStopError{Errors: errs}
+	}
+	return nil
+}
+
+func (ln *LocalNetwork) GetGenesisPath() string {
+	return filepath.Join(ln.Dir, "genesis.json")
+}
+
+func (ln *LocalNetwork) ReadGenesis() error {
+	bytes, err := os.ReadFile(ln.GetGenesisPath())
+	if err != nil {
+		return fmt.Errorf("failed to read genesis: %w", err)
+	}
+	genesis := genesis.UnparsedConfig{}
+	if err := json.Unmarshal(bytes, &genesis); err != nil {
+		return fmt.Errorf("failed to unmarshal genesis: %w", err)
+	}
+	ln.Genesis = &genesis
+	return nil
+}
+
+func (ln *LocalNetwork) WriteGenesis() error {
+	bytes, err := testnet.DefaultJSONMarshal(ln.Genesis)
+	if err != nil {
+		return fmt.Errorf("failed to marshal genesis: %w", err)
+	}
+	if err := os.WriteFile(ln.GetGenesisPath(), bytes, perms.ReadWrite); err != nil {
+		return fmt.Errorf("failed to write genesis: %w", err)
+	}
+	return nil
+}
+
+func (ln *LocalNetwork) GetChainConfigDir() string {
+	return filepath.Join(ln.Dir, "chains")
+}
+
+func (ln *LocalNetwork) GetCChainConfigPath() string {
+	return filepath.Join(ln.GetChainConfigDir(), "C", "config.json")
+}
+
+func (ln *LocalNetwork) ReadCChainConfig() error {
+	bytes, err := os.ReadFile(ln.GetCChainConfigPath())
+	if err != nil {
+		return fmt.Errorf("failed to read C-Chain config: %w", err)
+	}
+	cChainConfig := testnet.FlagsMap{}
+	if err := json.Unmarshal(bytes, &cChainConfig); err != nil {
+		return fmt.Errorf("failed to unmarshal C-Chain config: %w", err)
+	}
+	ln.CChainConfig = cChainConfig
+	return nil
+}
+
+func (ln *LocalNetwork) WriteCChainConfig() error {
+	path := ln.GetCChainConfigPath()
+	dir := filepath.Dir(path)
+	if err := os.MkdirAll(dir, perms.ReadWriteExecute); err != nil {
+		return fmt.Errorf("failed to create C-Chain config dir: %w", err)
+	}
+	bytes, err := testnet.DefaultJSONMarshal(ln.CChainConfig)
+	if err != nil {
+		return fmt.Errorf("failed to marshal C-Chain config: %w", err)
+	}
+	if err := os.WriteFile(path, bytes, perms.ReadWrite); err != nil {
+		return fmt.Errorf("failed to write C-Chain config: %w", err)
+	}
+	return nil
+}
```

Comment:
Wondering if we should be doing something generic here rather than being specific to the C-chain